pkg/kubelet/kubelet_node_status_test.go

```diff
@@ -747,71 +750,125 @@ func TestUpdateNodeStatusWithRuntimeStateError(t *testing.T) {
+				// Inject error into fake runtime status check, node should be NotReady
+				fakeRuntime.StatusErr = fmt.Errorf("injected runtime status error")
+				clock := testKubelet.fakeClock
+				clock.SetTime(time.Now())
+			},
+		},
+		{
+			Desc:   "Should report node not ready if runtime status is nil",
+			Reason: "KubeletNotReady",
+			Status: v1.ConditionFalse,
+			UpdateRuntimeFn: func(testKubelet *TestKubelet) {
+				fakeRuntime := testKubelet.fakeRuntime
+				fakeRuntime.RuntimeStatus = nil
+			},
+		},
+		{
+			Desc:   "Should report node not ready if runtime status is empty",
+			Reason: "KubeletNotReady",
+			Status: v1.ConditionFalse,
+			UpdateRuntimeFn: func(testKubelet *TestKubelet) {
+				fakeRuntime := testKubelet.fakeRuntime
+				fakeRuntime.RuntimeStatus = &kubecontainer.RuntimeStatus{}
+			},
+		},
+		{
+			Desc:   "Should report node not ready if RuntimeReady is false",
+			Reason: "KubeletNotReady",
+			Status: v1.ConditionFalse,
+			UpdateRuntimeFn: func(testKubelet *TestKubelet) {
+				fakeRuntime := testKubelet.fakeRuntime
+				fakeRuntime.RuntimeStatus = &kubecontainer.RuntimeStatus{
+					Conditions: []kubecontainer.RuntimeCondition{
+						{Type: kubecontainer.RuntimeReady, Status: false},
+						{Type: kubecontainer.NetworkReady, Status: true},
+					},
+				}
+			},
+		},
+		{
+			Desc:   "Should report node ready if RuntimeReady is true",
+			Reason: "KubeletReady",
+			Status: v1.ConditionTrue,
+			UpdateRuntimeFn: func(testKubelet *TestKubelet) {
+				fakeRuntime := testKubelet.fakeRuntime
+				fakeRuntime.RuntimeStatus = &kubecontainer.RuntimeStatus{
+					Conditions: []kubecontainer.RuntimeCondition{
+						{Type: kubecontainer.RuntimeReady, Status: true},
+						{Type: kubecontainer.NetworkReady, Status: true},
+					},
+				}
+			},
+		},
+		{
+			Desc:   "Should report node not ready if NetworkReady is false",
+			Reason: "KubeletNotReady",
+			Status: v1.ConditionFalse,
+			UpdateRuntimeFn: func(testKubelet *TestKubelet) {
+				fakeRuntime := testKubelet.fakeRuntime
+				fakeRuntime.RuntimeStatus = &kubecontainer.RuntimeStatus{
+					Conditions: []kubecontainer.RuntimeCondition{
+						{Type: kubecontainer.RuntimeReady, Status: true},
+						{Type: kubecontainer.NetworkReady, Status: false},
+					},
+				}
+			},
 		},
 	}
-	kubelet.updateRuntimeUp()
-	checkNodeStatus(v1.ConditionTrue, "KubeletReady")
 
-	// Should report node not ready if NetworkReady is false.
-	fakeRuntime.RuntimeStatus = &kubecontainer.RuntimeStatus{
-		Conditions: []kubecontainer.RuntimeCondition{
-			{Type: kubecontainer.RuntimeReady, Status: true},
-			{Type: kubecontainer.NetworkReady, Status: false},
-		},
+	for _, tc := range cases {
+		testKubelet := buildTestKubelet()
+		defer testKubelet.Cleanup()
+
+		kubelet := testKubelet.kubelet
+		if tc.UpdateRuntimeFn != nil {
+			tc.UpdateRuntimeFn(testKubelet)
+		}
+
+		kubelet.updateRuntimeUp()
+		checkNodeStatus(context.Background(), testKubelet, tc.Status, tc.Reason)
```

Comment:
Please use testing context instead of `context.Background()`, e.g.` tCtx := ktesting.init(t)`

Reply:
done